channeldb/graph.go

```diff
@@ -1525,12 +1526,10 @@ func (c *ChannelGraph) DisconnectBlockAtHeight(height uint32) ([]*ChannelEdgeInf
 		BlockHeight: height,
 	}
 
-	// Delete everything after this height from the db.
-	endShortChanID := lnwire.ShortChannelID{
-		BlockHeight: math.MaxUint32 & 0x00ffffff,
-		TxIndex:     math.MaxUint32 & 0x00ffffff,
-		TxPosition:  math.MaxUint16,
-	}
+	// Delete everything after this height from the db up until the
+	// SCID alias range.
+	endShortChanID := aliasmgr.StartingAlias
```

Comment:
What about permitting access to the actual scid range and existing set? Otherwise seems like this would be lossy and end up _not_ deleting some items that it actually should.

Reply:
Since these are zero-conf channels, I don't think they ever need to be removed from the graph in case of a reorg

Reply:
In the normal non-zero-conf case, I'm not sure how our code will handle a reorg removing the edge from the graph. I don't think another `ChannelAnnouncement` will be sent to the gossiper from the funding manager. If I'm right that means that the edge will just stay deleted similar to this case

Reply:
> In the normal non-zero-conf case, I'm not sure how our code will handle a reorg removing the edge from the graph. I don't think another `ChannelAnnouncement` will be sent to the gossiper from the funding manager. If I'm right that means that the edge will just stay deleted similar to this case

Apparently this is what the `testOpenChannelAfterReorg` does. I guess there's nothing we can do about it since the spec doesn't prescribe what to do if the funding tx is reorged out

Reply:
> Apparently this is what the testOpenChannelAfterReorg does. I guess there's nothing we can do about it since the spec doesn't prescribe what to do if the funding tx is reorged out

If we have an alias SCID for a non-zero-conf channel, I think we would still want the channel to be pruned from the graph on reorg (existing behavior).  Otherwise we might try to route payments through the non-existent channel.

Reply:
Non-zero-conf channels are not stored in the graph by their alias SCID. They still have an alias, but it is not stored in the graph this way. Only public zero-conf channels before 6 confirmations and private zero-conf channels will be stored in the graph via their alias SCID.